lib/rig_inbound_gateway/api_proxy/response_from_parser.ex

```diff
@@ -0,0 +1,103 @@
+defmodule RigInboundGateway.ApiProxy.ResponseFromParser do
+  @moduledoc """
+  Handles parsing of messages coming from different sources (e.g. Kafka)
+
+  """
+
+  require Logger
+
+  use Rig.Config, [:schema_registry_host]
+
+  alias Rig.Connection.Codec
+  alias RigKafka.Avro
+
+  @default_response_code 200
+
+  # ---
+
+  @spec parse([{String.t(), String.t()}, ...], map()) ::
+          {pid, pos_integer(), any(), map()} | String.t()
+  # structured
+  def parse(
+        _headers,
+        %{
+          "body" => body,
+          "rig" => rig_metadata
+        } = message
+      ) do
+    with {:ok, correlation_id} <- Map.fetch(rig_metadata, "correlation"),
+         {:ok, deserialized_pid} <- Codec.deserialize(correlation_id),
+         raw_response_code <- Map.get(rig_metadata, "response_code", @default_response_code),
```

Comment:
Not sure a default makes sense in this case. You have to provide that "rig" object anyway because of the correlation ID. Making the response code required at least prevents you from accidentally sending errors as OK - imagine you've accidentally misspelled "response_code" and didn't notice, because RIG just uses 200 anyway.

Reply:
Okay, wanted to make users life easier.